### PR TITLE
start the review ?

### PR DESCRIPTION
**Pull request by majeed-hm**
Hi @ivy-rew ,

I am done with the implementation. the only open topic is the GitHub build which still failing. 
can we start the review for this connector?

Best regards,
Majeed.

### CODE REVIEW

**Comment by alexsuter** [comment]
why don't we call this repo just "cronjob"

**Comment by ivy-rew** [comment]
> why don't we call this repo just "cronjob"

totally open for that; anyone disagreeing ?

File: cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	 */
+	public static class Editor extends UiEditorExtension {
+
+		private IUiFieldEditor globalVariable;
+
+		
+		@Override
+		public void initUiFields(ExtensionUiBuilder ui) {
+			ui.label("Global Variable name").create();
+			globalVariable = ui.textField().create();
```
Comment (ivy-rew) [inline cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java:182]: I think here is the best way to explain how this element works:
- > we should have a description field added via : ui.label("my explanatory text")
... here I'd leave a short word about where 'variables' are defined. And also how cron patterns are to be read.

File: cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	 * Editor class to work with the configuration.
+	 */
+	public static class Editor extends UiEditorExtension {
+
+		private IUiFieldEditor globalVariable;
+
+		
+		@Override
+		public void initUiFields(ExtensionUiBuilder ui) {
+			ui.label("Global Variable name").create();
```
Comment (ivy-rew) [inline cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java:181]: We no longer call that concept 'global variables' : just 'Variables' -> https://dev.axonivy.com/doc/10.0/engine-guide/reference/engine-cockpit/configuration.html#variables

**Comment by ivy-rew** [comment]
Hey @alexsuter ; can you make suggestions how the groupId+artifactId's in here should be? already good or should we change it?

File: cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+public class CronByGlobalVariableTriggerStartEventBean extends AbstractProcessStartEventBean implements Job {
+	private Scheduler scheduler = null;
+	private JobDetail job = null;
+	private CronTrigger trigger = null;
+	private String triggerIdentifier;
+	private static final String RUNTIME_KEY = "eventRuntime";
+	private static final Object SYN_OBJECT = new Object();
+	private static Map<String, Long> startedJobs = Collections.synchronizedMap(new HashMap<String, Long>());
+
+	/**
```
Comment (ivy-rew) [inline cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java:48]: I think that comment doesn't add any helpful information

File: cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		
+		@Override
+		public void initUiFields(ExtensionUiBuilder ui) {
+			ui.label("Global Variable name").create();
+			globalVariable = ui.textField().create();
+		}
+
+		@Override
+		public String getConfiguration() {
+			return globalVariable.getText();
```
Comment (ivy-rew) [inline cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java:187]: globalVariable -> variable

**Comment by ivy-rew** [comment]
Ok first version, ...we can integrate it as is: but should not publish any release. Until we have a test that runs and verifies the quality of the code in here.

**Review by ivy-rew — APPROVED** [review]
Thanks for your work @majeed-hm 
Looking forward to see this product in use by the Portal :+1:

**Comment by alexsuter** [comment]
We already have discussed the group id and I thought we said that it needs to called:

com.axonivy.utils.cronjob

I don't know why we need utils at the end?

artifact ids should be without utils too
cronjob
cronjob-demo
cronjob-test
...

**Comment by ivy-rew** [comment]
> 

thanks. I like your proposal. And agree... that 'util' doesn't add any information. So let's omit that part.

**Comment by majeed-hm** [comment]
Hi @ivy-rew ,

Thank you for your inputs. I updated the code accordingly but now the build still failing because of the following error:
`github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/reference/checks#create-a-check-run"}`
do you have an idea what could be the problem?

Regards,
Majeed.

**Comment by ivy-rew** [comment]
> Hi @ivy-rew ,
> 
> Thank you for your inputs. I updated the code accordingly but now the build still failing because of the following error: `github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/reference/checks#create-a-check-run"}` do you have an idea what could be the problem?
> 
> Regards, Majeed.

the build is very likely failing, as you are running in a very special environment. Did you check the file paths you are writing and reading from? Maybe /tmp isn't accessible as one would assume. Nevertheless, you could use java core apis to resolve the systems temp path und write/read into it. Then I'd guess the tests pass

**Comment by majeed-hm** [comment]
> > Hi @ivy-rew ,
> > Thank you for your inputs. I updated the code accordingly but now the build still failing because of the following error: `github.GithubException.GithubException: 403 {"message": "Resource not accessible by integration", "documentation_url": "https://docs.github.com/rest/reference/checks#create-a-check-run"}` do you have an idea what could be the problem?
> > Regards, Majeed.
> 
> the build is very likely failing, as you are running in a very special environment. Did you check the file paths you are writing and reading from? Maybe /tmp isn't accessible as one would assume. Nevertheless, you could use java core apis to resolve the systems temp path und write/read into it. Then I'd guess the tests pass

if it is not accessible then the UnitTest would fail before it comes to "Publish Unit Test Results".

**Comment by ivy-rew** [comment]
hi mr. support  @weissreto , can you give @majeed-hm  a hand here?

**Comment by alexsuter** [comment]
@majeed-hm seems that your repo is not properly configured.

I googled for you: publish-unit-test-result-action 403

https://seekdavidlee.medium.com/how-to-fix-post-repos-check-runs-403-error-on-github-action-workflow-f2c5a9bb67d

**Comment by majeed-hm** [comment]
> @majeed-hm seems that your repo is not properly configured.
> 
> I googled for you: publish-unit-test-result-action 403
> 
> https://seekdavidlee.medium.com/how-to-fix-post-repos-check-runs-403-error-on-github-action-workflow-f2c5a9bb67d

Thank you Alex, now it is working but I just used the https://github.com/axonivy-market/market-product as template and I didnt changed anything in the configuration and now I just created a test repo like I did before and the config is like the following 
image

so I assume a wrong config in the template.

**Comment by alexsuter** [comment]
The template is configured that way. But the settings won't be applied to the new repo. maybe we should mention that in the instruction notes.

**Comment by majeed-hm** [comment]
> The template is configured that way. But the settings won't be applied to the new repo. maybe we should mention that in the instruction notes.

Okay so now I think everything is ready for merge and close the PR?

**Comment by alexsuter** [comment]
fine for me

@ivy-rew let's merge?

**Pull request merged**